src/store/reducers/authReducer.ts

```diff
@@ -0,0 +1,45 @@
+import {AuthAction, AuthState, SET_USER, SET_LOADING, SIGN_OUT, SET_ERROR, SET_SUCCESS} from '../types';
+
+const initialState: AuthState = {
+  user: null,
+  authenticated: false,
+  loading: false,
+  error: '',
+  needVerification: false,
+  success: '',
+};
+
+export default (state = initialState, action: AuthAction) => {
+  switch (action.type) {
```

Comment:
Should have a better implementation for this (I am running out of idea for now tho), usually 
using `switch case` is not the best practice.

Reply:
I don't see how using a switch case isn't best practice most redux apps use switches to handle actions.

Reply:
Or this could be nested because imagine you have 30 actions there, it will be a looooong code to look at. (imagine 30 `case SOMETHING:`)

My first idea is that we could nest it or categorize the action type, and dispatch it or change it according to its type. (`User/Login` or something.)

Reply:
Just found something like this, it's like an object for mapping the handler according to the type.

https://medium.com/@jeremy_7548/making-the-case-for-redux-without-switch-statements-92635f1813b9

Reply:
That's why you separate your reducers into different files and combine them together through the index reducer file, authReducer is just for the authentication actions which at most is just signing in users, logging out users, and signing up users. If a reducer is having over 30 something cases that probably means we need to split it into a separate reducer file.

Reply:
Tho I still prefer using the factory pattern, this should be sufficient for now. Imma leave it there.